openpype/hosts/maya/plugins/publish/validate_xgen.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+import maya.cmds as cmds
+import xgenm
+
+import pyblish.api
+from openpype.pipeline.publish import KnownPublishError
+
+
+class ValidateXgen(pyblish.api.InstancePlugin):
+    """Validate Xgen data."""
+
+    label = "Validate Xgen"
+    order = pyblish.api.ValidatorOrder
+    host = ["maya"]
+    families = ["xgen"]
+
+    def process(self, instance):
+        set_members = instance.data.get("setMembers")
+
+        # Only 1 collection/node per instance.
+        if len(set_members) != 1:
+            raise KnownPublishError(
+                "Only one collection per instance is allowed."
+                " Found:\n{}".format(set_members)
+            )
+
+        # Only xgen palette node is allowed.
+        node_type = cmds.nodeType(set_members[0])
+        if node_type != "xgmPalette":
+            raise KnownPublishError(
+                "Only node of type \"xgmPalette\" are allowed. Referred to as"
+                " \"collection\" in the Maya UI."
+                " Node type found: {}".format(node_type)
```

Comment:
I believe these should be `KnownValidationError` or `PublishValidationError` or something.

Reply:
Got `PublishValidationError` and `KnownPublishError`. @iLLiCiTiT got any insights on this?

Reply:
`PublishValidationError` is for validation errors which can describe to user what happened, where and gives him option to run fix actions. `KnownPublishError` is for cases when publishing crashes but user can't do much about it.

Reply:
Thanks @iLLiCiTiT